firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -451,4 +470,34 @@ private Void deleteFirebaseInstallationId() throws FirebaseInstallationsExceptio
     persistedInstallation.insertOrUpdatePersistedInstallationEntry(entry.withNoGeneratedFid());
     return null;
   }
+
+  private static boolean isIIDVersionCompatible() {
+    try {
+      final Class<FirebaseInstanceId> firebaseInstanceIdClass = FirebaseInstanceId.class;
+      String iidVersion = getIIDVersion();
+      if (TextUtils.isEmpty(iidVersion)
+          || Utils.compareVersion(IID_SDK_VERSION_BEFORE_FIS, iidVersion) >= 0) {
+        return false;
+      }
+    } catch (NoClassDefFoundError exception) {
+      // There is no IID SDK at all.
+      return true;
+    }
+    return true;
+  }
+
+  private static String getIIDVersion() {
```

Comment:
@Nullable

Reply:
How about using Optional?

Reply:
Added the annotation. Can't add Optional - min SDK requirement for Optional is 24 and firebase sdk's are at 14